EIPS/eip-2381.md

```diff
@@ -0,0 +1,143 @@
+
+A single deployed EIP-721 contract can contain multiple non-fungible tokens. Tokens are addressed by an unsigned 256bit integer, `tokenID`. This EIP describes an extra getter and setter, `tokenID()` and `setTokenID()`, that are included in an ENS resolver contract, to allow for ENS names to resolve to a specific non fungible token within a given EIP-721 contract.
+
+This allows ENS domains like [`bugcat.cryptokitties.eth`](https://www.cryptokitties.co/kitty/101) and [`dragon.cryptokitties.eth`](https://www.cryptokitties.co/kitty/896775) to resolve to both the CryptoKitties contract address and the tokenID of the non-fungible kitty within it.
+
+## Motivation
+
+Non-fungible Tokens are meant to be globally unique, it makes sense to be able to name them. Giving NFTs names makes them more real to the non Ethereum enthusiast that doesn't understand what a contract address or tokenID is. A domain name in the existing web 2.0 world is already widely understood to be a finite resource, and as such, would help convey the scarcity that is a non-Fungible token to new users.
+
+This change might also to an extent, decentralise access to non-fungibles on Ethereum. Currently, the mapping for a human readable name to an NFT is kept off chain, typically on the platform of the NFT issuer. However, if the community moved towards naming NFTs on chain, using ENS, this would allow any client that supports ENS resolution to resolve an ENS name to an NFT, without querying the issuer's off-chain metadata.
+
+## Specification
+
+The ENS Resolver Profile specified by this EIP adds two functions to a deployed resolver contract:
+
+```
+    function tokenID(bytes32 node) public view returns(uint256);
+    function setTokenID(bytes32 node, uint256 token);
+```
+
+You can evaluate whether a given ENS resolver supports this EIP by using the EIP-165 standard `supportsInterface(bytes4)`.
+
+The interface identifier for this interface is `0x4b23de55`.
+
+## Rationale
+
+The design of this smart contract is a modified replica of the [existing contenthash resolver profile](https://github.com/ensdomains/resolvers/blob/master/contracts/profiles/ContentHashResolver.sol).
+
+The decision to use Contract Address and tokenID as a pair to uniquely identify EIP-721 NFTs is lifted directly from the [EIP-721 Rationale](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md#rationale) quoted here:
+
+    Every NFT is identified by a unique uint256 ID inside the EIP-721 smart contract. This identifying number SHALL NOT change for the life of the contract. The pair (contract address, uint256 tokenId) will then be a globally unique and fully-qualified identifier for a specific asset on an Ethereum chain.
```

Comment:
```suggestion
> Every NFT is identified by a unique uint256 ID inside the EIP-721 smart contract. This identifying number **SHALL NOT** change for the life of the contract. The pair (contract address, uint256 tokenId) will then be a globally unique and fully-qualified identifier for a specific asset on an Ethereum chain.
```
I'm not confident that an indent will result in a quote in all rendering engines, but I'm pretty sure `>` will.